doc/resourcemanagementbatch.rst

```diff
@@ -30,27 +35,129 @@ for details on getting a ``Credentials`` instance.
-provider with your subscription.
 
-Use the following code to do the registration. You can use the same
-credentials you created in the previous section.
+Create a Batch Account
+----------------------
+
+A Batch Account will need to be created in a specified location and resource group.
+The default Batch Account quota is 1 per location per subscription, but can be increased to a maximum of 50.
+Please contact support if you require a quota increase.
+For more information on resource groups and resource management, see :doc:`Resource Management<resourcemanagement>`.
+
+In order to make use of Application Packages, a storage account will need to be linked to the Batch Account. A linked storage account is known as 'auto-storage'.
+A storage account can be created with the :doc:`Storage Resource Management Client <resourcemanagementstorage>`.
 
 .. code:: python
 
-    from azure.mgmt.resource.resources import ResourceManagementClient, ResourceManagementClientConfiguration
+	# Create a Resource Group (or use an existing one)
+	import azure.mgmt.resource
+
+	RESOURCE_GROUP = 'python_sdk'
+	LOCATION = 'westus'
+
+	resource_client = = azure.mgmt.storage.ResourceManagementClient(
+		azure.mgmt.resources.ResourceManagementClientConfiguration(
+			credentials, # See section above
+			subscription_id
+		)
+	)
+	group = azure.mgmt.resource.resources.models.ResourceGroup(
+		name=RESOURCE_GROUP,
+		location=LOCATION
+	)
+	resource_client.resource_groups.create_or_update(
+		RESOURCE_GROUP,
+		group,
+	)
+
+
+	# Create a storage account for 'auto-storage' (or use an existing one)
+	import azure.mgmt.storage
+	storage_client = azure.mgmt.storage.StorageManagementClient(
+		azure.mgmt.storage.StorageManagementClientConfiguration(
+			credentials, # See section above
+			subscription_id
+		)
+	)
+	storage_params = azure.mgmt.storage.models.StorageAccountCreateParameters(
+		location=LOCATION,
+		account_type=azure.mgmt.storage.models.AccountType.standard_lrs
+	)
+	creating = storage_client.storage_accounts.create(
+		RESOURCE_GROUP,
+		'pythonstorageaccount',
+		storage_params
+	)
+	creating.wait()
+
+	# Create a Batch Account, specifying the storage account we want to link
+	storage_resource = '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Storage/storageAccounts/{}'.format(
+		subscription_id,
+		RESOURCE_GROUP,
+		'pythonstorageaccount'
+	)
+	batch_account = azure.mgmt.batch.models.BatchAccountCreateParameters(
+		location=LOCATION,
+		auto_storage=azure.mgmt.batch.models.AutoStorageBaseProperties(storage_resource)
+	)
+	creating = batch_client.account.create('MyBatchAccount', LOCATION, batch_account)
+	creating.wait()
+
+
+
+Account keys (used for authenticating the :doc:`Batch Client <batch>`) can be retrieved or regenerated.
 
-    resource_client = ResourceManagementClient(
-        ResourceManagementClient(
-            credentials,
-            subscription_id
-        )
-    resource_client.providers.register('Microsoft.Batch')
```

Comment:
@annatisch 
You remove this provider registration for you doc, it's not useful to do management?